app/models/accounting/interest_calculator.rb

```diff
@@ -59,8 +59,7 @@ def initialize(loan)
     end
 
     def recalculate
-      return if @loan.qb_division.qb_read_only
-      return unless @loan.txn_modification_allowed?
+      return unless TransactionPolicy.new(:machine, Transaction.new(project: @loan)).create?
```

Comment:
We pass `:machine` as the user b/c there is not really a user doing this action. Nothing says you have to pass a User as the user param :) The joys of Pundit!

Reply:
interesting!